python/grass/temporal/metadata.py

```diff
@@ -1134,7 +1135,8 @@ def __init__(
         self.D["ewres_min"] = None
         self.D["ewres_max"] = None
         self.D["aggregation_type"] = aggregation_type
-        self.D["number_of_bands"] = None
+        if get_tgis_db_version_from_metadata() > 2:
+            self.D["number_of_bands"] = None
```

Comment:
Awesome! I miss the context to understand how much of these you plan/need to add. In this particular case, it seems that it would not be a big deal to simply report number of bands as None (or zero or whatever this should translates to). It might be matter of taste and the impact on the user is questionable, but in my view, the library does not have to behave as the older version. In my view, number of bands is zero in 8.0.0 with data from 7.8. Where it should fail in this context is attempt to register bands and maybe in some cases also when explicitly accessing band information.

Reply:
> Awesome! I miss the context to understand how much of these you plan/need to add. In this particular case, it seems that it would not be a big deal to simply report number of bands as None (or zero or whatever this should translates to). It might be matter of taste and the impact on the user is questionable, but in my view, the library does not have to behave as the older version. In my view, number of bands is zero in 8.0.0 with data from 7.8. 

I think it can also be 0 with data from 8.0.0 if we do not register bands... 

> Where it should fail in this context is attempt to register bands and maybe in some cases also when explicitly accessing band information.

So, a warning would be needed then only if a user tries to register or read bands in an old STRDS (i.e. version 2) while in GRASS 8.0. But s/he could still read info from the old STRDS and do operations with it? 

Honestly, I think it's really easy and safe to do a `t.upgrade` in all mapsets with STDS if using grass-dev or once 8.0.0 is released... So, as long as #1435 removes the errors when registering raster maps without band references, I do not have a strong opinion about backward compatibility.

Reply:
I would keep the `None` value, it's default for other values, see https://github.com/landam/grass/blob/tgisdb3_backward_compatibility/python/grass/temporal/metadata.py#L1129 and https://github.com/landam/grass/blob/tgisdb3_backward_compatibility/python/grass/temporal/metadata.py#L1145. Actually the `None` value is retured by `get_number_of_bands` (https://github.com/landam/grass/blob/tgisdb3_backward_compatibility/python/grass/temporal/metadata.py#L1234).